### PR TITLE
[ISSUE #7543] Retry topic v2 in pop

### LINKED ISSUE #7543: [Enhancement] Pop retry topic v2
### Before Creating the Enhancement Request

- [X] I have confirmed that this should be classified as an enhancement rather than a bug/feature.

### Summary

目前在pop机制中使用的retry topic的格式是形如 "%RETRY%group_topic"，其中分隔符"_"是一个合法字符，可以被使用在topic或者group中，因此会导致两个问题。

1. pop retry topic重复，比如由group_a，topic组成的retry topic，和由group，a_topic组成的retry topic均为%RETRY%group_a_topic
2. 由于采用了合法字符"_"作为分隔符，因此没办法根据retry topic反向得出group和topic，在消息通知和需要拆分group的场景都会存在问题

因此本方案提出了pop retry topic v2，以"+"为分割符，即形如"%RETRY%group+topic",其中"+"为topic和group无法使用的字符，因此解决了上述两个问题，同时"+"也隐式的表达了订阅的含义。

Currently, in the pop mechanism, the format used for the retry topic is as the form of "%RETRY%group_topic". The underscore character "_" serves as a valid separator, which can be used within both the topic and group. Consequently, this situation leads to two issues:

1. Repetition of pop retry topics: For instance, a retry topic composed of group_a and topic, and another retry topic composed of group and a_topic, both result in "%RETRY%group_a_topic".
2. Due to the utilization of the underscore character "_" as a separator, it becomes impossible to deduce the group and topic from the retry topic. This presents challenges in scenarios involving message notifications and the need to split the group.

Therefore, this proposal suggests the implementation of pop retry topic v2, employing the "+" as the separator. The new format would appear as "%RETRY%group+topic". Since "+" is not a permissible character in either topics or groups, it effectively resolves the aforementioned issues. Furthermore, the use of "+" implicitly expresses the subscription meaning.

### Motivation

Fix the issue of original pop retry topic.

### Describe the Solution You'd Like

Retry topic v2 as the format of "%RETRY%group+topic" where "+" is the separator.

### Describe Alternatives You've Considered

No

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by drpmma**
### Which Issue(s) This PR Fixes

Fixes #7543

### Brief Description

Use new retry topic to solve pop retry topic message notification and split group from retry topic.

### How Did You Test This Change?

Unit test.

### CODE REVIEW

**Comment by xdkxlk** [comment]
We need to consider the modifications from  #7208 in order to ensure compatibility.

**Review by lollipopjin — APPROVED** [review]
LGTM

**Review by joeCarf — COMMENTED** [review]
> We need to consider the modifications from #7208 in order to ensure compatibility.

+1， it will only delete the v2 topic instead of the v1 topic

**Review by RongtongJin — APPROVED** [review]

File: common/src/main/java/org/apache/rocketmq/common/BrokerConfig.java [hunk 1]
```diff
@@ -223,6 +223,8 @@ public class BrokerConfig extends BrokerIdentity {
     private boolean enablePopBatchAck = false;
     private boolean enableNotifyAfterPopOrderLockRelease = true;
     private boolean initPopOffsetByCheckMsgInMem = true;
+    // read message from pop retry topic v1, for the compatibility, will be removed in the next version
+    private boolean retrieveMessageFromPopRetryTopicV1 = true;
```
Comment (RongtongJin) [inline common/src/main/java/org/apache/rocketmq/common/BrokerConfig.java:227]: If retrieveMessageFromPopRetryTopicV1 is later removed, some ecosystem-related proxies (such as MQTT proxy) will need to be synchronized and modified accordingly.

**Comment by drpmma** [comment]
> We need to consider the modifications from #7208 in order to ensure compatibility.

fixed

**Comment by leizhiyuan** [comment]
Can the forward slash be used for folders in Linux? The "topic" of Pop ultimately ends up as a folder on the machine？

Or is it talking about the modification that relies on rocksdb?

**Review by joeCarf — APPROVED** [review]
LGTM

**Comment by drpmma** [comment]
> Can the forward slash be used for folders in Linux? The "topic" of Pop ultimately ends up as a folder on the machine？
> 
> Or, in other words, is the design utilizing a multi-level directory to support this type of topic?

Change separator from '/' to ':' to prevent multi-level directory case

**Review by RongtongJin — APPROVED** [review]

**Pull request merged**